File: Epargne/Models/DTOs/ClientDTO.cs
namespace Epargne.Models.DTOs;

/// <summary>
/// DTO pour les informations client
/// Structure identique aux autres modules
/// </summary>
public class ClientDTO
{
    public long? IdClient { get; set; }
    public string? NumeroClient { get; set; }
    public string Nom { get; set; } = string.Empty;
    public string Prenom { get; set; } = string.Empty;
    public DateTime DateNaissance { get; set; }
    public string NumCin { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Telephone { get; set; }
    public string? Adresse { get; set; }
    public string? CodePostal { get; set; }
    public string? Ville { get; set; }
    public string? Profession { get; set; }
    public decimal? RevenuMensuel { get; set; }
    public decimal SoldeInitial { get; set; }
    public string? SituationFamiliale { get; set; }
    public string? MotDePasse { get; set; } // Pour inscription/modification
    public string? Statut { get; set; }
    public DateTime? DateCreation { get; set; }
    public DateTime? DateModification { get; set; }
}

File: Pret/src/main/java/com/banque/pret/dto/ClientDTO.java
package com.banque.pret.dto;

import lombok.*;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * DTO pour l'entité Client
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ClientDTO implements Serializable {
    
    private static final long serialVersionUID = 1L;

    private Long idClient;
    private String numeroClient;
    private String nom;
    private String prenom;
    private LocalDate dateNaissance;
    private String numCin;
    private String email;
    private String telephone;
    private String adresse;
    private String codePostal;
    private String ville;
    private String profession;
    private BigDecimal revenuMensuel;
    private BigDecimal soldeInitial;
    private String situationFamiliale;  // String au lieu de l'enum
    private String motDePasse;
    private String statut;  // String au lieu de l'enum
    private LocalDateTime dateCreation;
    private LocalDateTime dateModification;
}
